src/ajax.js

```diff
@@ -605,6 +611,12 @@ jQuery.extend( {
 					cacheURL + ( rquery.test( cacheURL ) ? "&" : "?" ) + "_=" + nonce++;
 			}
 
+			// Put hash back on the actual URL that will be requested (gh-1732)
+			s.url = cacheURL + hash;
+
+			// cacheURL doesn't have the anti-cache param or hash
+			cacheURL = cacheURL.replace( rts, "" );
```

Comment:
The final value of `cacheURL` only matters if `s.ifModified` is true, but I think it should **include** `_` parameters that come from the user rather than us. What if we updated like so?

```js
uncached = s.url.slice( cacheURL.length );

if ( s.data ) {
	cacheURL += ( rquery.test( cacheURL ) ? "&" : "?" ) + s.data;
	…
}

if ( s.cache === false ) {
	cacheURL = cacheURL.replace( rts, "" );
	uncached = ( rquery.test( cacheURL ) ? "&" : "?" ) + "_=" + (nonce++) + uncached;
}

s.url = cacheURL + uncached;
```